apps/src/lab2/types.ts

```diff
@@ -91,6 +100,34 @@ export interface UpdateSourceOptions extends SaveSourceOptions {
 // Blockly JSON is currently typed as a generic object
 export type BlocklySource = {[key: string]: unknown};
 
+// -- SKETCH LAB -- //
+
+export type SketchlabExternalFiles = Record<FileId, SketchlabProjectFile>;
+
+// By default, Excalidraw file entries require a dataURL field that has a
+// base64 encoding of the file. As we move to store images in S3, this field
+// is now optional.
+type ExcalidrawFileWithOptionalData = Omit<BinaryFileData, 'dataURL'> & {
+  dataURL?: DataURL;
+};
+
+type ExcalidrawFilesWithOptionalData = Record<
+  ExcalidrawElement['id'],
+  ExcalidrawFileWithOptionalData
+>;
+
+export type ExcalidrawSourceWithExternalFiles = Omit<
+  ExcalidrawInitialDataState,
+  'files'
+> & {
+  files?: ExcalidrawFilesWithOptionalData;
```

Comment:
Could we add a comment here about why we need both `files` and `externalFiles`? This comment from https://github.com/code-dot-org/code-dot-org/pull/69199#discussion_r2478934634 was helpful!

Reply:
Yep good call, updated!